yt_dlp/extractor/plex.py

```diff
@@ -0,0 +1,471 @@
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    determine_ext,
+    float_or_none,
+    int_or_none,
+    parse_age_limit,
+    traverse_obj,
+)
+
+
+class PlexWatchBaseIE(InfoExtractor):
+    _NETRC_MACHINE = 'plex'
+    # provider url is in https://plex.tv/media/providers?X-Plex-Token=<plex_token>
+    # nb: need Accept: application/json, otherwise return xml
+    _CDN_ENDPOINT = {
+        'vod': 'https://vod.provider.plex.tv',
+        'live': 'https://epg.provider.plex.tv',
+        # from api ( identifier : baseUrl)
```

Comment:
?

Reply:
I mean the first two endpoint is to simpler the api request in watch.plex and the other is mainly used in app.plex and can found in api `https://plex.tv/media/providers?X-Plex-Token=<plex_token>` with identifier as key and baseUrl as value